### PR TITLE
feat(EMS-3373): release please updated to v4

### PR DESCRIPTION
**Pull request by abhi-markan**
## Introduction :pencil2:
Ensure Google release please [deprecated plugin](https://github.com/google-github-actions/release-please-action) has been replaced with the [recommended script](https://github.com/googleapis/release-please)

## Resolution :heavy_check_mark:
* Replaced GHA plugin with `release-please` CLI command.
* Added few additional GHA `github.yml` and `pr.yml`.
* Added prettier check in `lint.yml`.

## Miscellaneous :heavy_plus_sign:
* Dependencies updates
* Updated `database` docker compose service command to `--mysql-native-password=ON`.
* Added `cookie-parser` UI `test-modes`.
* Fix/update an E2E test with date assertion failures.

### CODE REVIEW

File: .husky/commit-msg [hunk 1]
```diff
@@ -1,6 +1,4 @@
 #!/usr/bin/env sh
 . "$(dirname -- "$0")/_/husky.sh"
 
-npx commitlint --edit $1
-npx lint-staged
-
+npx commitlint --color --edit $1
```
Comment (ttbarnes) [inline .husky/commit-msg:4]: Should `npx lint-staged` remain here?

Comment (abhi-markan) [reply .husky/commit-msg:4]: Moved to `pre-commit`

**Review by Zainzzkk — DISMISSED** [review]

File: package.json [hunk 2]
```diff
@@ -21,60 +21,63 @@
     "lint": "npm i --legacy-peer-deps && npm i --legacy-peer-deps --prefix ./src/ui && eslint --ext .ts src/ui && npm i --legacy-peer-deps --prefix ./src/api && eslint --ext .ts src/api && eslint --ext .js e2e-tests",
     "lint:fix": "prettier --write src/ui  && eslint --ext .ts src/api --fix && eslint --ext .ts src/ui --fix &&  eslint --ext .js e2e-tests --fix",
     "lint:fix:all": "prettier --write src/ui  && eslint --ext .ts src/api --fix && eslint --ext .ts src/ui --fix &&  eslint --ext .js e2e-tests --fix",
+    "prettier": "prettier --no-error-on-unmatched-pattern --log-level error --check **/*.{ts,js}",
```
Comment (ttbarnes) [inline package.json:24]: I don't believe we need this, since `prettier` is run alongside the `lint` commands?

Comment (abhi-markan) [reply package.json:24]: I have left the `prettier` npm script however temporarly disabled the check in `lint.yml` as I first need to unify `prettier` and `ESLint` configurations.

File: src/ui/server/test-mocks/index.ts [hunk 3]
```diff
@@ -35,6 +36,7 @@ const mockReq = () => {
       origin: 'https://mock.com',
       host: 'mock-host.com',
     },
+    signedCookies: cookieParser(''),
```
Comment (ttbarnes) [inline src/ui/server/test-mocks/index.ts:39]: Is this required? Doesn't seem so, `signedCookies` isn't used anywhere?

Comment (abhi-markan) [reply src/ui/server/test-mocks/index.ts:39]: Without this I had Typescript error.

File: src/ui/package.json [hunk 4]
```diff
@@ -22,43 +22,43 @@
...
   },
   "dependencies": {
-    "@babel/core": "^7.23.7",
-    "@babel/plugin-transform-runtime": "^7.23.7",
-    "@babel/preset-env": "^7.23.7",
-    "@ministryofjustice/frontend": "^2.0.0",
+    "@babel/core": "^7.24.5",
+    "@babel/plugin-transform-runtime": "^7.24.3",
+    "@babel/preset-env": "^7.24.5",
+    "@ministryofjustice/frontend": "^2.1.3",
```
Comment (Zainzzkk) [inline src/ui/package.json:28]: we need to rememeber to remove this when merging main into no-pdf

**Review by ttbarnes — DISMISSED** [review]

**Review by ttbarnes — APPROVED** [review]

**Review by Zainzzkk — APPROVED** [review]

**Review by ttbarnes — APPROVED** [review]

**Pull request merged**